tests/providers/google/cloud/hooks/test_compute_ssh.py

```diff
@@ -162,7 +162,9 @@ def test_get_conn_authorize_using_instance_metadata(
         mock_compute_hook.return_value.project_id = TEST_PROJECT_ID
         mock_compute_hook.return_value.get_instance_address.return_value = EXTERNAL_IP
 
-        mock_compute_hook.return_value.get_instance_info.return_value = {"metadata": {}}
+        mock_compute_hook.return_value.get_instance_info.return_value = {
+            "metadata": {"items": [{"key": "foo", "value": "bar"}]}
```

Comment:
Can we please also test other scenarios?
1. When no items are there.
example - expected value: `user:pubkey\n`
3. When the key is present in items, we should end up with the right key. 
example - expected value: `user:pubkey\n<ssh-key-in-items>`

Something like this PR - https://github.com/apache/airflow/pull/37190/files

Reply:
Definitely, I will use your PR as a guide to update ASAP

Reply:
I added your test cases to this PR, can you check it? @utkarsharma2

Reply:
The new test case looks good. I think we don't need any changes to the old one, WDYT?

Reply:
Yes, I just missed that. I reverted this change and pushed it